src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendRecorder.java

```diff
@@ -249,31 +257,56 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
                             if(updateBuildResult(build, profile, errorMessage)) {
                                 return true;
                             } else {
+                                errorEncountered = true;
                                 continue;
                             }
                         } else {
                             condition.setApplicationId(app.getId());
-                            appId = condition.getApplicationId();
-                            VulnerabilityTrendHelper.logMessage(listener, "Fetched Application : [name = '" + condition.getApplicationOriginName() + "', displayName = '" + app.getName() + "', agentType='" + app.getLanguage() + "'] with ID: [" + appId + "]");
+                            VulnerabilityTrendHelper.logMessage(listener, "Fetched Application : [name = '" + condition.getApplicationOriginName() + "', displayName = '" + app.getName() + "', agentType='" + app.getLanguage() + "'] with ID: [" + condition.getPreparedApplicationId() + "]");
                         }
                     } catch (UnauthorizedException e) {
-                        String errorMessage = CONTRAST_ERROR_PREFIX + "Unable to retrieve application information from TeamServer.";
+                        String errorMessage = CONTRAST_ERROR_PREFIX + "Unable to retrieve application information from Contrast.";
                         VulnerabilityTrendHelper.logMessage(listener, errorMessage);
                         VulnerabilityTrendHelper.logMessage(listener, e.getMessage());
                         if(updateBuildResult(build, profile, errorMessage)) {
                             return true;
                         } else {
+                            errorEncountered = true;
                             continue;
                         }
                     }
                     break;
                 case APPLICATION_ID:
                 default:
-                    if (VulnerabilityTrendHelper.getAppIdFromAppTitle(appId) != null) {
-                        appId = VulnerabilityTrendHelper.getAppIdFromAppTitle(appId);
-                    }
                     break;
             }
+        }
+
+        try {
+            if (globalThresholdRequired(thresholdConditions, contrastSDK, profile)) {
+                thresholdConditions =
+                    VulnerabilityTrendHelper.getThresholdConditions(conditions, globalThresholdConditions);
+                if (thresholdConditions.isEmpty()) {
+                    String errorMessage = CONTRAST_ERROR_PREFIX + "Vulnerability Security Controls for connection '" + profile.getName() + "' are not defined.";
+                    VulnerabilityTrendHelper.logMessage(listener, errorMessage);
+                    if(updateBuildResult(build, profile, errorMessage)) {
+                        return true;
+                    } else {
+                        errorEncountered = true;
```

Comment:
Can we add some log in updateBuildResult or here to notify that profile.isApplyVulnerableBuildResultOnContrastError() is set to false? (Maybe we already do this somewhere else?) That way you could get a log like, you have a required global threshold, but your security controls are not defined, BUT we aren't failing loudly because you told us not to.

Reply:
done